src/lib/memory/tracking_memory_resource.cpp

```diff
@@ -0,0 +1,26 @@
+#include "tracking_memory_resource.hpp"
+
+namespace opossum {
+
+void* TrackingMemoryResource::do_allocate(std::size_t bytes, std::size_t alignment) {
+  auto timestamp = std::chrono::system_clock::now();
```

Comment:
Be aware that the `system_clock` may be adjusted during run-time. It is not guaranteed that the timestamps you get when calling `system_clock::now()` are (i) ordered and (ii) have the correct temporal distance from each other. In your case, whenn deallocating memory, you could get a timestamp that lies before the one you git when allocating the memory. When one does not need to map timestamps to wall-clock time, it is recommended to use `std::chrono::steady_clock`. See #2423 for details.